src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNetWebApi2Integration.cs

```diff
@@ -25,7 +25,12 @@ public static class AspNetWebApi2Integration
         /// <returns>A task with the result</returns>
         public static object ExecuteAsync(object apiController, object controllerContext, object cancellationTokenSource)
         {
-            var cancellationToken = ((CancellationTokenSource)cancellationTokenSource).Token;
+            if (apiController == null) { throw new ArgumentNullException(nameof(apiController)); }
+
+            if (controllerContext == null) { throw new ArgumentNullException(nameof(controllerContext)); }
```

Comment:
I'd be in favor of not throwing exceptions to make sure we interfere as little as possible with the normal program execution. Maybe we could log something instead.

Reply:
Those arguments can't actually be null in normal program execution. `ExecuteAsync()` is an instance method and `apiController` is the `this` argument.

I removed the null check for `controllerContext`, since we only use it to populate the span (we already avoid using it if it is null). If it is null, [Web API will throw quickly](https://github.com/aspnet/AspNetWebStack/blob/749384689e027a2fcd29eb79a9137b94cea611a8/src/System.Web.Http/ApiController.cs#L574) later on.

I left the check for `apiController` since we actually need that value and we cannot move forward without it. Also, `ArgumentNullException` looks better and provides more information than the inevitable `NullReferenceException`.